### PR TITLE
[frontend-api] Made functional tests independent on domain locales

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Current state of functional tests in [frontend api](https://github.com/shopsys/frontend-api) is not suitable for changing locales on domain. When doing so the FEAPI tests will start failing. This PR will make the tests more stable
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php [hunk 1]
```diff
@@ -112,15 +113,16 @@ private function getArticleDataProvider(): array
     /**
      * @return array
      */
-    public function getSpecialArticleDataProvider(): array
+    private function getSpecialArticleDataProvider(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:118]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticlesTest.php [hunk 2]
```diff
@@ -202,46 +203,47 @@ private function getLastArticlesQuery(int $numberOfArticles, ?string $placement
      */
     private function getExpectedArticles(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Article/GetArticlesTest.php:206]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Category/CategoriesTest.php [hunk 3]
```diff
@@ -18,14 +19,15 @@ public function testRootCategories(): void
             }
         ';
 
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Category/CategoriesTest.php:22]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Category/CategoriesTest.php [hunk 4]
```diff
@@ -46,15 +48,16 @@ public function testChildCategories(): void
...
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Category/CategoriesTest.php:51]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Customer/User/CurrentCustomerUserTest.php [hunk 5]
```diff
@@ -83,10 +84,11 @@ public function testChangePersonalDataWithWrongData(): void
     }
 }';
 
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Customer/User/CurrentCustomerUserTest.php:87]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/AbstractOrderTestCase.php [hunk 6]
```diff
@@ -16,9 +17,10 @@ class AbstractOrderTestCase extends GraphQlTestCase
      */
     protected function getExpectedOrderItems(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/AbstractOrderTestCase.php:20]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/DeliveryFieldsAreValidatedTest.php [hunk 7]
```diff
@@ -4,44 +4,47 @@
 
 namespace Tests\FrontendApiBundle\Functional\Order;
 
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+
 class DeliveryFieldsAreValidatedTest extends AbstractOrderTestCase
 {
     public function testValidationErrorWhenCompanyBehalfIsTrueAndFieldsAreMissing(): void
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/DeliveryFieldsAreValidatedTest.php:13]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/FullOrderTest.php [hunk 8]
```diff
@@ -4,20 +4,23 @@
 
 namespace Tests\FrontendApiBundle\Functional\Order;
 
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+
 class FullOrderTest extends AbstractOrderTestCase
 {
     public function testCreateFullOrder(): void
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/FullOrderTest.php:13]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsAuthenticatedCustomerUserTest.php [hunk 9]
```diff
@@ -52,14 +53,15 @@ public function testGetOrderReturnsError(): void
      */
     private function getOrderDataForCurrentlyLoggedCustomerUserProvider(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsAuthenticatedCustomerUserTest.php:56]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsAuthenticatedCustomerUserTest.php [hunk 10]
```diff
@@ -52,14 +53,15 @@ public function testGetOrderReturnsError(): void
...
         $data = [];
         $orderIds = [1, 2, 3];
         foreach ($orderIds as $orderId) {
             $order = $this->orderFacade->getById($orderId);
             $data[] = [
                 $order->getUuid(),
                 [
-                    'status' => $order->getStatus()->getName(),
+                    'status' => $order->getStatus()->getName($firstDomainLocale),
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsAuthenticatedCustomerUserTest.php:64]: This should not be necessary as tests are run always on first domain and its locale is set as default in translation entities

File: project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php [hunk 11]
```diff
@@ -53,14 +54,15 @@ public function testGetOrderReturnsError(): void
...
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
         $data = [];
         $orderIds = [7, 8, 9];
         foreach ($orderIds as $orderId) {
             $order = $this->orderFacade->getById($orderId);
             $data[] = [
                 $order->getUrlHash(),
                 [
-                    'status' => $order->getStatus()->getName(),
+                    'status' => $order->getStatus()->getName($firstDomainLocale),
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php:65]: Same as previous

File: project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php [hunk 12]
```diff
@@ -53,14 +54,15 @@ public function testGetOrderReturnsError(): void
      */
     public function getOrderDataProvider(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php:57]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/GetOrdersAsAuthenticatedCustomerUserTest.php [hunk 13]
```diff
@@ -135,29 +135,30 @@ private function getLastOrdersQuery(int $numberOfOrders): string
      */
     private function getExpectedUserOrders(): array
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/GetOrdersAsAuthenticatedCustomerUserTest.php:138]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/MinimalOrderTest.php [hunk 14]
```diff
@@ -4,20 +4,23 @@
 
 namespace Tests\FrontendApiBundle\Functional\Order;
 
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+
 class MinimalOrderTest extends AbstractOrderTestCase
 {
     public function testCreateMinimalOrderMutation(): void
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/MinimalOrderTest.php:13]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/MultipleProductsInOrderTest.php [hunk 15]
```diff
@@ -5,21 +5,23 @@
 namespace Tests\FrontendApiBundle\Functional\Order;
 
 use App\DataFixtures\Demo\ProductDataFixture;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
 
 class MultipleProductsInOrderTest extends AbstractOrderTestCase
 {
     public function testCreateFullOrder(): void
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/MultipleProductsInOrderTest.php:14]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

File: project-base/tests/FrontendApiBundle/Functional/Order/MultipleProductsInOrderTest.php [hunk 16]
```diff
@@ -5,21 +5,23 @@
...
                     ],
                     'payment' => [
-                        'name' => 'Cash on delivery',
+                        'name' => t('Cash on delivery', [], 'dataFixtures', $firstDomainLocale),
                     ],
-                    'status' => 'New',
+                    'status' => t('New [adjective]', [], 'dataFixtures', $firstDomainLocale),
                     'totalPrice' => [
                         'priceWithVat' => '4964.44',
                         'priceWithoutVat' => '4103.19',
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Order/MultipleProductsInOrderTest.php:27]: You will also need to updated all prices in all tests like this:
```suggestion
                        'priceWithVat' => $this->getPriceWithVatConvertedToDomainDefaultCurrency('4964.44'),
                        'priceWithoutVat' => $this->getPriceWithVatConvertedToDomainDefaultCurrency('4103.19'),
```

File: project-base/tests/FrontendApiBundle/Functional/Product/ProductsTest.php [hunk 17]
```diff
@@ -160,13 +160,14 @@ public function testLastProduct(): void
      */
     private function getExpectedDataForLastProduct(): string
     {
+        $firstDomainLocale = $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID)->getLocale();
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Product/ProductsTest.php:163]:
```suggestion
        $firstDomainLocale = $this->getLocaleForFirstDomain();
```

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php [hunk 18]
```diff
@@ -48,32 +48,31 @@ public function testGetArticleReturnsError(): void
...
-     * @param array $expectedData
-     * @dataProvider getSpecialArticleDataProvider
-     */
-    public function testGetSpecialArticle(string $graphQlType, array $expectedData): void
+    public function testGetSpecialArticle(): void
     {
-        $response = $this->getResponseContentForQuery($this->getSpecialArticleQuery($graphQlType));
-        $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
-        $responseData = $this->getResponseDataForGraphQlType($response, $graphQlType);
+        foreach ($this->getSpecialArticleDataProvider() as $dataSet) {
```
Comment (grossmannmartin) [inline project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:53]: why remove @dataprovider and wrap the whole method into foreach?

Comment (s3tezsky) [reply project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:53]: The dataprovider is dependent on property set via `setUp` method. Sadly phpunit uses dataprovider sooner than `setUp` is called so it end on `calling * on null`

**Review by grossmannmartin — CHANGES_REQUESTED** [review]
Hi,

I found few locale-related tests that fail, can you please check them out?

```
1) Tests\FrontendApiBundle\Functional\Order\GetOrderAsAuthenticatedCustomerUserTest::testGetOrder
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'Done'
+'Vyřízena'

/Volumes/work/shopsys/framework/project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsAuthenticatedCustomerUserTest.php:28

2) Tests\FrontendApiBundle\Functional\Order\GetOrderAsUnauthenticatedCustomerUserTest::testGetOrder
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'Canceled'
+'Stornována'

/Volumes/work/shopsys/framework/project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php:28

3) Tests\ReadModelBundle\Functional\Product\Listed\ListedProductViewFacadeTest::testGetAllAccessories
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Kabel HDMI A - HDMI A M/M 2m zlacene konektory High Speed HD'
+'Kabel HDMI A - HDMI A M/M 2m gold-plated connectors High Speed HD'

/Volumes/work/shopsys/framework/project-base/tests/ReadModelBundle/Functional/Product/Listed/ListedProductViewFacadeTest.php:40
```

**Comment by s3tezsky** [comment]
That is because you have not changed `default_locale`. See https://docs.shopsys.com/en/latest/introduction/how-to-set-up-domains-and-locales/#38-default-application-locale

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php (hunk 18 above)
Comment (grossmannmartin) [reply project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:53]: Sad, but I understand.

**Review by TomasLudvik — APPROVED** [review]

**Review by sslt — APPROVED** [review]

**Pull request merged**